.github/workflows/tests_linux_x86.yml

```
@@ -27,21 +27,48 @@ jobs:
       max-parallel: 1
 
     steps:
+      - name: Checkout pennyLane-lightning-gpu
+        uses: actions/checkout@v3
+
       - uses: actions/setup-python@v4
+        id: setup_python
         name: Install Python
         with:
           python-version: '3.8'
 
-      - name: Remove Ubuntu unattended upgrades
-        run: sudo apt-get remove -y -q unattended-upgrades
+      # Since the self-hosted runner can be re-used. It is best to set up all package
+      # installations in a virtual environment that gets cleaned at the end of each workflow run
+      - name: Setup Python virtual environment
+        id: setup_venv
+        env:
+          VENV_NAME: ${{ github.workspace }}/venv_${{ steps.setup_python.outputs.python-version }}_${{ github.sha }}
+        run: |
+          # Clear any pre-existing venvs
+          #rm -rf venv_*
```

```suggestion
          rm -rf venv_*
```
Best to leave this in